### DISCUSSION TITLE
Using unregister on array item

### DISCUSSION THREAD

**Discussion by MImranAsghar** [react-hook-form/react-hook-form#6412] [Q&A]
Hey, I have an array registered for the hook form, using register('arr[0]'), when I unregister that using unregister('arr[0]'), and console log values from getValues(), I see that the 0th index is now equal to undefined after unregister but it is not gone totally as I had expected. Is this the expected behaviour and if so is there a way to get around that so the 0th index is removed and I have an empty array in this example?

**Comment by MImranAsghar** [comment]
@bluebill1049 what do you think?

**Reply by MImranAsghar** [reply]
For example, after I unregister 0th item in the array, I get this where item 0 becomes undefined but is not removed
![image](https://user-images.githubusercontent.com/12305084/131535329-d2afab84-00da-4125-b59c-0c168ca047ef.png)

**Reply by MImranAsghar** [reply]
But I expected item 1 to have become item 0 and prev item 0 to have been removed (also, if it's the version issue where if this issue is already fixed in future version of the library, I would be unable to make version change on app as it would currently require lots of other changes for now. In which case, I was wondering if there is a workaround with keeping the same version of react-hook-form?)

**Comment by bluebill1049** [comment]
it would be difficult if you map the item and running the `register` function again unless you unmount and remove that input completely.
